sdks/python/apache_beam/io/localfilesystem.py

```diff
@@ -321,11 +321,22 @@ def _delete_path(path):
         raise IOError(err)
 
     exceptions = {}
-    for path in paths:
+
+    def try_delete(path):
       try:
         _delete_path(path)
       except Exception as e:  # pylint: disable=broad-except
         exceptions[path] = e
 
+    for match_result in self.match(paths):
+      metadata_list = match_result.metadata_list
+
+      if not metadata_list:
```

Comment:
Why do you want to make this an error? (does another filesystem do this or a comment say this is the right contract?)

I would suspect that empty matches turn into no-ops.

Reply:
Turns out that we test that delete throws an error if non existent and that we have an inconsistent definition between multiple implementations (e.g Java LocalFileSystem). Filed https://issues.apache.org/jira/browse/BEAM-5425 to ask whether the contract should be changed to allow prior partial successes to continue and for errors to not be thrown for non-existent files.

Reply:
sorry I dropped this; yea I only made it an error to conform to the existing test you mentioned